src/main/java/com/vestmark/bitbucket/plugin/VersionComparator.java

```diff
@@ -52,12 +52,35 @@ public int compare(T o1, T o2)
         result = leftInt.compareTo(rightInt);
       }
       else {
+        // If we made it here, there is at least one string in the comparison.
+        // Example, 8.1 versus ubs-8.0.10
```

Comment:
please remove client name

Reply:
Unfortunately, I just discovered that is does not address the issue for PRs made in a <text>-8.0.10 type branch.  The tool thinks ALL other release branches are downstream from it.
It's the same logic that tells a PR made in 8.0.9, for instance, that <text>-8.0.10 is upstream from it, therefore causing it not to appear in the merge path from 8.0.9 forward.
The compare method in the VersionComparator is doing double duty.  It is called for sorting and also to determine which branches are upstream from each other, and I can't have it both ways.  
Needs more work :-(

Reply:
The comparator is not the (only) issue - it's the collection that needs to be adjusted (upstream).